zkvm/src/predicate.rs

```diff
@@ -144,36 +125,25 @@ impl Predicate {
     }
 }
 
-impl PredicateWitness {
-    fn to_uncompressed_point(&self) -> RistrettoPoint {
-        match self {
-            PredicateWitness::OpaqueBranch(p) => *p,
-            PredicateWitness::Key(s) => VerificationKey::from_secret_uncompressed(s),
-            PredicateWitness::Or(l, r) => {
-                let l = l.to_uncompressed_point();
-                let r = r.to_uncompressed_point();
-                let f = Predicate::commit_or(l.compress(), r.compress());
-                l + f * PedersenGens::default().B
-            }
-            PredicateWitness::Program(prog) => {
-                let mut bytecode = Vec::new();
-                Instruction::encode_program(prog.iter(), &mut bytecode);
-                let h = Predicate::commit_program(&bytecode);
-                h * PedersenGens::default().B_blinding
-            }
-        }
+impl Into<CompressedRistretto> for Predicate {
+    fn into(self) -> CompressedRistretto {
+        self.to_point()
     }
 }
 
-impl From<CompressedRistretto> for Predicate {
-    fn from(p: CompressedRistretto) -> Self {
-        Predicate::Opaque(p)
-    }
-}
+impl PredicateDisjunction {
+    pub fn new(left: Predicate, right: Predicate) -> Result<Self, VMError> {
```

Comment:
We probably don't need to use `PredicateDisjunction::new` in other contexts, so how about:

1. Removing `PredicateDisjunction::new` method.
2. Doing its job directly in the `Predicate::or` method.